test/Cardano/FeeSpec.hs

```diff
@@ -460,6 +466,54 @@ propDistributeFeeIsFair (fee, coins) = (.&&.)
         * fromIntegral (getFee fee)
         % fromIntegral (sum $ getCoin <$> coins)
 
+-- | Verify that fees are distributed optimally across coins, such that the
+-- absolute deviation from the ideal (unrounded) fee distribution is minimal.
+propDistributeFeeOptimal
+    :: Fee
+    -> NonEmpty Coin
+    -> Property
+propDistributeFeeOptimal fee coins = property $
+    computeDeviation (fst <$> distributeFee fee coins)
+        `shouldBe` minimumPossibleDeviation
+  where
+    -- Compute the deviation of a fee portion distribution from the ideal
+    -- unrounded fee portion distribution.
+    computeDeviation :: NonEmpty Fee -> Rational
+    computeDeviation feesRounded = F.sum $
+        NE.zipWith (\(Fee f) u -> abs (fromIntegral f - u))
+            feesRounded
+            idealUnroundedDistribution
+
+    -- The minimum deviation across all possible distributions for the given
+    -- fee and set of coins.
+    minimumPossibleDeviation :: Rational
+    minimumPossibleDeviation =
+        F.minimum $ computeDeviation <$> allPossibleDistributions
+
+    -- The set of all possible fee distributions for the given fee and coins.
+    allPossibleDistributions :: [NonEmpty Fee]
+    allPossibleDistributions = filter isValidDistribution $ NE.zipWith
+        (\f roundDir -> Fee $ fromIntegral @Integer $ round roundDir f)
```

Comment:
ok, so the idea is to have ideal distribution, and then *perturb* it using all possible roundings, and then filter out valid distributions.

Reply:
> ok, so the idea is to have ideal distribution, and then **apply perturbation** it using all possible roundings, and then filter out valid distributions.

Yes, exactly!